### ISSUE TITLE
Failed to open the uploaded document

### ISSUE THREAD

**Issue by AK47Poi** [DeepLcom/deepl-python#117]
Hi,when I use the code to upload the file for translation, the translation reports an error.How can I change my code?
`deepl.exceptions.DocumentTranslationException: Error occurred while translating document: Failed to open the uploaded document.`

**Comment by JanEbbing** [comment]
This indicates an error on the server side (the server could not open your document). Does this happen consistently, even if you retry the translation? Are you sending a valid document of the [supported types](https://developers.deepl.com/docs/api-reference/document)?

**Comment by AK47Poi** [comment]
"I encounter the same error when I retry the translation, and I am uploading files of types docx, xlsx, and pptx. Can I provide the docid to help investigate the issue? Here it is: F2916EC7142D837C30197CD40147E58C."

**Comment by AK47Poi** [comment]
I encounter the same error when I retry the translation, and I am uploading files of types docx, xlsx, and pptx. Can I provide the docid to help investigate the issue?

**Comment by fernandolvwrk** [comment]
To solve this problem, do not use the file path, send the file as a binary data stream.
```
with open(file=filepath, mode="rb") as f:
	doc_handle = translator.translate_document_upload(input_document=f.read(),... 
```

**Comment by JanEbbing** [comment]
Were you using `translate_document` on a filepath? That method only works with file-like objects, as the docs state. We have `translate_document_from_filepath` to pass in file paths. See for example our tests on these:

[From filepath](https://github.com/DeepLcom/deepl-python/blob/main/tests/test_translate_document.py#L16)

[From file object](https://github.com/DeepLcom/deepl-python/blob/main/tests/test_translate_document.py#L81)

**Comment by fernandolvwrk** [comment]
Yes, you are 100% right. `translate_document_from_filepath()` works fine with a filepath, but my app needs to execute the steps individually, so I had to use `translate_document_upload()`, `translate_document_get_status()`, and `translate_document_download()`. You are right again, the docs state clearly that translate_document_upload() requires a file object or a stream. Thanks!
